### PR TITLE
Fix importing single-image Eiger files

### PR DESCRIPTION
**Pull request by benjaminhwilliams**
Currently the logic for importing a single-image Eiger file is broken.  This fixes that and introduces a test of `FormatHDF5EigerNearlyNexus` that uses a smaller data set hosted in https://github.com/dials/data & https://github.com/dials/data-files.

Test depends on dials/data#161.

### CODE REVIEW

**Comment by benjaminhwilliams** [comment]
As soon as we get dials/data#161 merged, I'll mark this as ready for review.

**Comment by benjaminhwilliams** [comment]
OK @dagewa, this is ready for review.

**Comment by dagewa** [comment]
Thanks @benjaminhwilliams, I added @graeme-winter as a reviewer too, particularly for the logic regarding single image imports, which is something he has been considering recently

**Comment by benjaminhwilliams** [comment]
Oops, sorry for the misleading commit message, I got my wires crossed.  I just added a news fragment.

**Comment by graeme-winter** [comment]
```
――――――――――――――――――――――――― tests/test_image_readers.py ――――――――――――――――――――――――――
ImportError while importing test module '/Users/graeme/svn/cctbx/modules/dxtbx/tests/test_image_readers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Users/graeme/svn/cctbx/modules/dxtbx/tests/test_image_readers.py:20: in <module>
    from dxtbx.format.image import (
E   ImportError: cannot import name 'CBFReader'
```

=> 🤔

**Comment by graeme-winter** [comment]
Verified that reading single image set works with the changes in here, fails on master

File: tests/format/test_FormatHDF5EigerNearlyNexus.py [hunk 1]
```diff
@@ -11,6 +11,47 @@
 pytest.importorskip("h5py")
 
 
+def test_semi_synthetic_dectris_eiger_nearly_nexus(dials_data, tmpdir):
+    master_h5 = dials_data("image_examples").join("dectris_eiger_master.h5").strpath
```
Comment (graeme-winter) [inline tests/format/test_FormatHDF5EigerNearlyNexus.py:15]: Welcome the addition of a test here 🙂

File: format/nexus.py [hunk 2]
```diff
@@ -1085,74 +1085,45 @@ def find_scanning_axis(obj):
             return o
 
 
-class ScanFactory(object):
+def generate_scan_model(obj, detector_obj):
```
Comment (graeme-winter) [inline format/nexus.py:1088]: Onions -= 1 \o/

**Comment by Anthchirp** [comment]
The work here was now re-merged in #164.

**Pull request merged**